README.rst

```diff
@@ -0,0 +1,320 @@
+  no data is lost when a thread fails.
+* All messages in queues are serialized by ``pickle`` which was for us a reason
+  against using this library.
+
+Kafka
+-----
+* For our needs as local system with IPC it's an unnecessary overhead.
+* Integration of Kafka written in Scala and Java in our C++/Python/Go codebase
+  not that simple.
+* Hard to setup and maintain `Kafka <https://kafka.apache.org/>`_.
+* broker system eventually a bottleneck.
+
+RabbitMQ
+--------
+* For our needs as local system with IPC it's an unnecessary overhead.
+* broker system eventually a bottleneck.
+* `RabbitMQ <https://www.rabbitmq.com//>`_ is less scalable than Kafka, but is
+  supported officially by more languages.
+
+zeromq persistence pattern
+--------------------------
+* Only persistence pattern of `zeromq <http://zeromq.org//>`_ is
+  `Titanic <https://rfc.zeromq.org/spec:9/TSP//>`_ which is also a broker
+  system. This takes away the purpose and advantage of zeromq to be a
+  lightweight library which requires no broker.
+
+Usage
+=====
+
+The usage of the library can be differentiated into deployment and actual
+runtime of your processes.
+
+Deployment
+----------
+
+For deployment only a proper config JSON file is needed to setup the whole
+queue structure.
+
+config.json
+^^^^^^^^^^^
+Publishers, subscribers and queues need to be defined before runtime in the
+config JSON file.
+
+.. code-block:: json
+
+    {
+        "pub": {
+            "out_queue": "/home/user/queues/queue",
+            "subscribers": ["sub"]
+        },
+        "sub": {
+            "in_queue": "/home/user/queues/queue"
+        },
+        "queues": {
+            "/home/user/queues/queue": {
+                "path": "/home/user/queues/queue",
+                "max_reader_num": 1024,
+                "max_db_num": 1024,
+                "max_db_size_bytes": 34359738368,
+                "subscribers": ["sub"],
+                "high-water-mark": {
+                    "MSG_TIMEOUT_SECS": 600,
+                    "MAX_MSGS_NUM": 10000,
+                    "HWM_LMDB_SIZE_BYTES": 1000000,
+                    "strategy": "prune_first"
+                }
+            }
+        }
+    }
+
+Control
+^^^^^^^
+
+Initialize all queues
+"""""""""""""""""""""
+
+.. code-block:: python
+
+    import persipubsub.control
+
+    persipubsub.control.initialize_all_dbs(config_pth="/home/user/config.json")
+
+Prune all dangling messages
+"""""""""""""""""""""""""""
+
+.. code-block:: python
+
+    import persipubsub.control
+
+    persipubsub.control.prune_dangling_messages(
+        config_pth="/home/user/config.json")
+
+Clear all messages
+""""""""""""""""""
+
+.. code-block:: python
+
+    import persipubsub.control
+
+    persipubsub.control.clear_all_subs(config_pth="/home/user/config.json")
+
+
+Runtime
+-------
+During runtime only publisher and subscriber are needed. Control can be
+optionally be used for pruning although the queues prune itself on a regular
```

Comment:
for pruning. Note: the queues prune themselves when the high water mark is reached (?)